File: app/src/main/java/locus/api/mapper/ImageDataConverter.java
package locus.api.mapper;

import androidx.annotation.Nullable;

import com.arcao.geocaching.api.data.ImageData;

import locus.api.objects.geocaching.GeocachingImage;

final class ImageDataConverter {
    @Nullable
    GeocachingImage createLocusGeocachingImage(@Nullable ImageData imageData) {
        if (imageData == null)
            return null;

        GeocachingImage image = new GeocachingImage();
        image.setName(imageData.name());
        image.setDescription(imageData.description());
        image.setThumbUrl(imageData.thumbUrl());
        image.setUrl(imageData.url());

        return image;
    }
}

File: app/src/main/java/locus/api/mapper/ImageDataConverter.kt
package locus.api.mapper

import androidx.annotation.Nullable
import com.arcao.geocaching4locus.data.api.model.Image
import locus.api.objects.geocaching.GeocachingImage

class ImageDataConverter {
    @Nullable
    fun createLocusGeocachingImage(@Nullable imageData: Image?): GeocachingImage? {
        if (imageData == null)
            return null

        return GeocachingImage().apply {
            name = imageData.description.orEmpty()
            description = imageData.description.orEmpty()
            thumbUrl = imageData.thumbnailUrl.orEmpty()
            url = imageData.url
        }
    }
}
